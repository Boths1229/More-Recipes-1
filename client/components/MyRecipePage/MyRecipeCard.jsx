/* eslint-disable no-unused-vars */
/**
 *  @fileOverview Creates header for authenticated users
 *
 *  @author       Sanni Taiwo
 *
 *  @requires     NPM:react
 */
import React, { Component } from 'react';
import img from '../../public/images/recipe-5.jpg';

/**
 * @description A class to create card for each recipes
 *
 * @extends Component
 */
class MyRecipeCard extends Component {
  /**
   * @description binds the method to get current recipe
   *
   * @param {object} props
   *
   * @returns {void}
  */
  constructor(props) {
    super(props);
    this.getRecipeDetails = this.getRecipeDetails.bind(this);
  }

  /**
   * @description get the current recipe
   *
   * @param {object} event
   *
   * @memberof MyRecipeCard
   *
   * @returns {void}
  */
  getRecipeDetails() {
    this.props.getRecipe(this.props.index);
  }

  /**
   * @description renders the component
   *
   * @param {object} event
   *
   * @memberof MyRecipeCard
   *
   * @returns {void} returns the component
  */
  render() {
    const { index } = this.props;
    return (
      <div className='col-md-4 card-space'>
        <div className='card-deck'>
          <article className='card'>
            <img src={img} className='card-img-top img-fluid' />
            <div className='card-body'>
              <h3 className='card-title'>
                <a href="/recipes/:recipeId" id="remove-link">
                  {this.props.recipeDetails.recipeName}
                </a>
              </h3>
            </div>
            <div >
              <div style={{ display: 'inline-flex', marginTop: '10px' }}>
                <div>
                  <button
                    data-toggle="modal"
                    data-target="#editModal"
                    className='btn edit-btn'
                    onClick={this.getRecipeDetails}
                    type='button'>
                    <i className="fa fa-pencil-square-o fa-lg" aria-hidden="true">
                      <span style={{ marginLeft: '5px', marginRight: '5px' }}></span>
                    </i>
                  </button>
                </div>
                <div style={{ marginLeft: '20px' }}>
                  <button
                    data-toggle="modal"
                    data-target="#deleteModal"
                    className='btn btn-danger'
                    onClick={this.getRecipeDetails}
                    type='button'>
                    <i className="fa fa-trash-o fa-lg" aria-hidden="true">
                      <span style={{ marginLeft: '5px', marginRight: '5px' }}></span>
                    </i>
                  </button>
                </div>
              </div>
            </div>
          </article>
        </div>
      </div>
    );
  }
}

export default MyRecipeCard;
