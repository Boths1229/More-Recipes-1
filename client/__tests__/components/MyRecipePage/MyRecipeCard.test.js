import React from 'react';
import { shallow } from 'enzyme';
import MyRecipeCard from '../../../components/MyRecipePage/MyRecipeCard';
import MockData from '../../__mocks__/actions/recipes';

let props;

const setup = () => {
  props = {
    index: 2,
    getRecipe: () => {},
    recipeDetails: MockData.recipeData
  };

  return shallow(<MyRecipeCard {...props} />);
};

describe('MyRecipeCard Component', () => {
  it('should render correctly', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('h3').length).toBe(1);
    expect(wrapper.find('div').length).toBe(8);
  });

  it('should get Details of recipe to be deleted', () => {
    const wrapper = setup();
    const action = wrapper.instance();
    const spy = jest.spyOn(action, 'getRecipeDetails');
    action.getRecipeDetails();
    expect(spy).toBeCalled();
  });
});
