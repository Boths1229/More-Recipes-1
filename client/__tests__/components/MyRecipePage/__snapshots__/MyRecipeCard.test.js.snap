// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyRecipeCard Component should render correctly 1`] = `
<div
  className="col-md-4 card-space"
>
  <div
    className="card-deck"
  >
    <article
      className="card recipe-card"
    >
      <div
        className="my-recipe-img"
      >
        <img
          alt="recipe"
          className="card-img-top img-fluid recipe-card-img"
          src="https://firebasestorage.googleapis.com/v0/b/more-recipes-3be20.appspot.com/o/images%2F702446f3-a2c2-48c7-a9d1-af20be9281d7.jpeg?alt=media&token=1827adb2-4108-4939-9fe0-778e27dd5212"
        />
      </div>
      <div
        className="card-body"
      >
        <h3
          className="card-title"
        >
          <Link
            id="remove-link"
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/recipes/27"
          >
            new recipe
          </Link>
        </h3>
      </div>
      <div>
        <div
          style={
            Object {
              "display": "inline-flex",
              "marginTop": "10px",
            }
          }
        >
          <div>
            <button
              className="btn edit-btn"
              data-for="edit"
              data-target="#editModal"
              data-tip={true}
              data-toggle="modal"
              onClick={[Function]}
              type="button"
            >
              <i
                aria-hidden="true"
                className="fa fa-pencil-square-o fa-lg"
              />
            </button>
            <ReactTooltip
              id="edit"
              insecure={true}
              place="left"
              resizeHide={true}
              type="dark"
              wrapper="div"
            >
              <span>
                Edit Recipe
              </span>
            </ReactTooltip>
          </div>
          <div
            style={
              Object {
                "marginLeft": "20px",
              }
            }
          >
            <button
              className="btn btn-danger"
              data-for="delete"
              data-target="#deleteModal"
              data-tip={true}
              data-toggle="modal"
              onClick={[Function]}
              type="button"
            >
              <i
                aria-hidden="true"
                className="fa fa-trash-o fa-lg"
              />
            </button>
            <ReactTooltip
              id="delete"
              insecure={true}
              place="right"
              resizeHide={true}
              type="dark"
              wrapper="div"
            >
              <span>
                Delete Recipe
              </span>
            </ReactTooltip>
          </div>
        </div>
      </div>
    </article>
  </div>
</div>
`;
